In [2]:
import scipy.stats as st
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sn
import statistics

In [3]:
pd.set_option('display.max.rows',None)
df2 = pd.read_csv('Laliga.csv')
df2

Unnamed: 0           Unnamed: 1 Unnamed: 2 Unnamed: 3   Unnamed: 4  \
0         Pos                 Team    Seasons     Points  GamesPlayed   
1           1          Real Madrid         86       4385         2762   
2           2            Barcelona         86       4262         2762   
3           3      Atletico Madrid         80       3442         2614   
4           4             Valencia         82       3386         2664   
5           5      Athletic Bilbao         86       3368         2762   
6           6              Sevilla         73       2819         2408   
7           7             Espanyol         82       2792         2626   
8           8        Real Sociedad         70       2573         2302   
9           9             Zaragoza         58       2109         1986   
10         10           Real Betis         51       1884         1728   
11         11  Deportivo La Coruna         45       1814         1530   
12         12           Celta Vigo         51       1789         1698   
13         13           Valladolid         42       1471         1466   
14         14     Racing Santander         44       1416         1428   
15         15       Sporting Gijon         43       1389         1458   
16         16              Osasuna         37       1351         1318   
17         17               Malaga         36       1314         1255   
18         18               Oviedo         38       1174         1192   
19         19             Mallorca         27       1148          988   
20         20           Las Palmas         33       1020         1096   
21         21           Villarreal         17        970          646   
22         22              Granada         23        667          742   
23         23       Rayo Vallecano         17        662          652   
24         24                Elche         21        606          678   
25         25               Getafe         12        553          456   
26         26             Hercules         20        538          628   
27         27             Tenerife         13        510          494   
28         28               Murcia         18        445          586   
29         29               Alaves         12        421          380   
30         30              Levante         11        416          402   
31         31            Salamanca         12        375          423   
32         32             Sabadell         14        353          426   
33         33                Cadiz         12        343          448   
34         34             Logrones          9        293          346   
35         35            Castellon         11        285          334   
36         36             Albacete          7        277          270   
37         37              Almeria          6        242          228   
38         38              Cordoba          9        230          282   
39         39           Compostela          4        190          160   
40         40           Recreativo          5        188          186   
41         41            Burgos CF          6        168          204   
42         42           Pontevedra          6        150          180   
43         43             Numancia          4        148          152   
44         44                Eibar          3        132          114   
45         45               Arenas          7        107          130   
46         46          Real Burgos          3         96          114   
47         47            Gimnastic          4         91          116   
48         48          Extremadura          2         83           80   
49         49               Merida          2         81           80   
50         50             Alcoyano          4         76          108   
51         51                 Jaen          3         71           90   
52         52           Real Union          4         56           72   
53         53           AD Almeria          2         52     

In [4]:
df2.columns

Index(['Unnamed: 0', 'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4',
       'Unnamed: 5', 'Unnamed: 6', 'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9',
       'Unnamed: 10', 'Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13',
       'Unnamed: 14', 'Unnamed: 15', 'Unnamed: 16', 'Unnamed: 17',
       'Unnamed: 18', 'Unnamed: 19'],
      dtype='object')

1. You were recently hired as a business analyst in a top sports company. The senior management team has asked you to come up with metrics with which they can gauge which team will win the upcoming La Liga cup (Football tournament). The given data set contains information on all the teams that have so far participated in all the past tournaments. It has data about how many goals each team scored, conceded; how many times they came within the first 6 positions, how many seasons they have qualified, their best position in the past, etc. You are required to do the following:

Before doing any analysis it would be a good idea to check for any hypens or other symbols the data set and make appropriate replacements to make sure you can perform arithmetic operations on the data. Prepare a short report to answer the following questions:
1.	Which are the teams which started playing between 1930-1980?
2.	Which are the top 5 teams in terms of points?
3.	What is the distribution of the winning percentage for all teams? Which teams are in the top 5 in terms of winning percentage? (Winning percentage= (GamesWon / GamesPlayed)*100)
4.	Is there a significant difference in the winning percentage for teams which have attained a best position between 1-3 and those teams which have had a best position between 4-7.


1. Which are the teams which started playing between 1930-1980?

In [25]:
team_debut = df2.loc[:,['Team','Debut']]
df_team_year1 = team_debut[team_debut['Debut']>='1930']
df_team_year2 = df_team_year1[df_team_year1['Debut']<='1980']
df_team_year2

Team    Debut
3              Valencia  1931-32
5               Sevilla  1934-35
8              Zaragoza  1939-40
9            Real Betis  1932-33
10  Deportivo La Coruna  1941-42
11           Celta Vigo  1939-40
12           Valladolid  1948-49
14       Sporting Gijon  1944-45
15              Osasuna  1935-36
16               Malaga  1949-50
17               Oviedo  1933-34
18             Mallorca  1960-61
19           Las Palmas  1951-52
21              Granada  1941-42
22       Rayo Vallecano  1977-78
23                Elche  1959-60
25             Hercules  1935-36
26             Tenerife  1961-62
27               Murcia  1940-41
28               Alaves  1930-31
29              Levante  1963-64
30            Salamanca  1974-75
31             Sabadell  1943-44
32                Cadiz  1977-78
34            Castellon  1941-42
37              Cordoba  1962-63
39           Recreativo  1978-79
40            Burgos CF  1971-72
41           Pontevedra  1963-64
46            Gimnastic  1947-48
49             Alcoyano  1945-46
50                 Jaen  1953-54
52           AD Almeria  1979-80
54               Lleida  1950-51
57               Condal  1956-57
58      Atletico Tetuan  1951-52
59     Cultural Leonesa  1955-56

2.Which are the top 5 teams in terms of points?

In [26]:
team_points = df2.loc[:,['Team','Points']]
team_points.sort_values(by=['Points'],ascending=False).head()

Team  Points
0      Real Madrid  4385.0
1        Barcelona  4262.0
2  Atletico Madrid  3442.0
3         Valencia  3386.0
4  Athletic Bilbao  3368.0

3.What is the distribution of the winning percentage for all teams? Which teams are in the top 5 in terms of winning percentage? (Winning percentage= (GamesWon / GamesPlayed)*100)

In [27]:
Played_Won = df2.loc[:,['Team','GamesPlayed','GamesWon']]
Win_Percentage = Played_Won['GamesWon']/Played_Won['GamesPlayed']*100
Win_Percentage.head()   

0    59.630702
1    57.241130
2    47.475134
3    44.557057
4    43.772629
dtype: float64

4.Is there a significant difference in the winning percentage for teams which have attained a best position between 1-3 and those teams which have had a best position between 4-7.

In [28]:
Played_Won1 = df2.loc[:,['Pos','Team','GamesPlayed','GamesWon']]
Win_Pos123 = Played_Won1[Played_Won1['Pos'].isin([1,2,3])]
#Win_Pos123
Win_Percentage1 = Win_Pos123['GamesWon']/Win_Pos123['GamesPlayed']*100
Win_Percentage1.head().sum()   

164.3469659002076

In [29]:
Played_Won2 = df2.loc[:,['Pos','Team','GamesPlayed','GamesWon']]
Win_Pos4567 = Played_Won2[Played_Won2['Pos'].isin([4,5,6,7])]
#Win_Pos4567
Win_Percentage2 = Win_Pos4567['GamesWon']/Win_Pos4567['GamesPlayed']*100
Win_Percentage2.head().sum() 

165.54317552797508

In [30]:
Win_Percentage2.head().sum() - Win_Percentage1.head().sum()  
# There is 1.19% difference from Teams in Position 4,5,6,7 to Teams in Position 1,2,3

1.1962096277674732

2. A study was done to measure the blood pressure of 60 year old women with glaucoma. A random sample of 200 60-year old women with glaucoma was chosen. The mean of the systolic blood pressure in the sample was 140 mm Hg and the standard deviation was 25 mm Hg.

a)	Calculate the estimated standard error of the sample mean? What does the standard error indicate?

In [31]:
n = 200
xbar = 140
s = 25
se = s/np.sqrt(n)
se

1.7677669529663687

b)	Estimate a 95% confidence interval for the true mean blood pressure for all 60-year old women with glaucoma.

In [32]:
alpha = 0.05
talpha_by_2 = st.t.isf(alpha/2,n-1)
lower = xbar-talpha_by_2*se
upper = xbar+talpha_by_2*se
print(lower,upper)

136.51404038839016 143.48595961160984


c)	Assume that instead of 200, a random sample of only 100 60-year old women with glaucoma was chosen. The sample mean and standard deviation estimates are the same as those in the original study. What is the estimated 95% confidence interval for the true mean blood pressure?

In [33]:
n1 = 100
xbar = 140
s = 25
alpha = 0.05
t_alpha_by_2 = st.t.isf(alpha/2,n1-1)
lower1 = xbar-t_alpha_by_2*se
upper1 = xbar+t_alpha_by_2*se
print(lower1,upper1)

136.49236684560728 143.50763315439272


d)	Which of the two above intervals are wider?

In [34]:
print("The Interval 1 for n= 200 is",upper - lower)
print("The Interval 2 for n= 100 is",upper1 - lower1)
print("Hence the Interval 2 is wider than Interval 1")

The Interval 1 for n= 200 is 6.971919223219686
The Interval 2 for n= 100 is 7.015266308785442
Hence the Interval 2 is wider than Interval 1


e)	Explain in non-technical terms why the estimated standard error of a sample mean tends to decrease with an increase in sample size.

The size (n) of a statistical sample affects the standard error for that sample. Because n is in the denominator of the standard error formula, the standard error decreases as n increases. It makes sense that having more data gives less variation 


3. Par Inc., is a major manufacturer of golf equipment. Management believes that Par’s market share could be increased with the introduction of a cut-resistant, longer-lasting golf ball. Therefore, the research group at Par has been investigating a new golf ball coating designed to resist cuts and provide a more durable ball. The tests with the coating have been promising.

One of the researchers voiced concern about the effect of the new coating on driving distances. Par would like the new cut-resistant ball to offer driving distances comparable to those of the current-model golf ball. To compare the driving distances for the two balls, 40 balls of both the new and current models were subjected to distance tests. The testing was performed with a mechanical hitting machine so that any difference between the mean distances for the two models could be attributed to a difference in the design. The results of the tests, with distances measured to the nearest yard, are contained in   the data set “Golf”.

In [35]:
df1 = pd.read_csv('Golf.csv')
df1

Current  New
0       264  277
1       261  269
2       267  263
3       272  266
4       258  262
5       283  251
6       258  262
7       266  289
8       259  286
9       270  264
10      263  274
11      264  266
12      284  262
13      263  271
14      260  260
15      283  281
16      255  250
17      272  263
18      266  278
19      268  264
20      270  272
21      287  259
22      289  264
23      280  280
24      272  274
25      275  281
26      265  276
27      260  269
28      278  268
29      275  262
30      281  283
31      274  250
32      273  253
33      263  260
34      275  270
35      267  263
36      279  261
37      274  255
38      276  263
39      262  279

1.	Formulate and present the rationale for a hypothesis test that Par could use to compare the driving distances of the current and new golf balls.

In [36]:
# H0: uc = ue, # H1: uc != ue


270.275
267.5


2.	Analyze the data to provide the hypothesis testing conclusion. What is the p-value for your test? What is your recommendation for Par Inc.?

In [68]:
n = 40
mean_current = df1.loc[:,'Current'].mean()
print(mean_current)
mean_new = df1.loc[:,'New'].mean()
print(mean_new)

270.275
267.5


In [69]:
n1= 40
S1= df1.loc[:,'Current']
sample1=list(S1)
print(sample1)
std1 = statistics.stdev(sample1) 
se1 = std1/np.sqrt(n1)
print(se1,std1)

[264, 261, 267, 272, 258, 283, 258, 266, 259, 270, 263, 264, 284, 263, 260, 283, 255, 272, 266, 268, 270, 287, 289, 280, 272, 275, 265, 260, 278, 275, 281, 274, 273, 263, 275, 267, 279, 274, 276, 262]
1.3839684207898637 8.752984838884595


In [70]:
n2 = 40
S2= df1.loc[:,'New']
sample2=list(S2)
print(sample2)
std2 = statistics.stdev(sample2)
se2 = std2/np.sqrt(n2)
print(se2,std2)

[277, 269, 263, 266, 262, 251, 262, 289, 286, 264, 274, 266, 262, 271, 260, 281, 250, 263, 278, 264, 272, 259, 264, 280, 274, 281, 276, 269, 268, 262, 283, 250, 253, 260, 270, 263, 261, 255, 263, 279]
1.5648379944000428 9.896904462947894


In [71]:
se = (std1+std2)/np.sqrt(n1+n2)
se

2.0851210125871766

In [103]:
t_test = (mean_current - mean_new)/se
print(t_test)
p_value = st.t.cdf(t_test,n-1)*2
print(p_value,alpha)
# As p_value > alpha hence Accept Null Hypothesis (Fail to Reject)

1.3308580093185156
1.8090355149887036 0.05


3.	What is the 95% confidence interval for the population mean of each model, and what is the 95% confidence interval for the difference between the means of the two population?

In [114]:
# Current
n1 = 40
alpha = 0.05
t_alpha_by_2 = st.t.isf(alpha/2,n1-1)
lower_cur = mean_current-t_alpha_by_2*se1
upper_cur = mean_current+t_alpha_by_2*se1
print(lower_cur,upper_cur,mean_current)

267.47565965314044 273.0743403468595 270.275


In [115]:
# New
n2 = 40
alpha = 0.05
t_alpha_by_3 = st.t.isf(alpha/2,n2-1)
lower_new = mean_new-t_alpha_by_3*se2
upper_new = mean_new+t_alpha_by_3*se2
print(lower_new,upper_new,mean_new)

264.3348164103898 270.6651835896102 267.5


There is 95%probability that the true population mean is between 267.47565965314044 and 273.0743403468595 incurrent model while 264.3348164103898 and 270.6651835896102 in new model.And  by  their  confidence  intervals,  we  can  say  that  there  is  no  significance  difference between the driving distances of Current Model and New Model.
